include/picongpu/plugins/openPMD/openPMDWriter.hpp

```diff
@@ -954,11 +954,13 @@ namespace openPMD
         void
         initWrite()
         {
-            mThreadParams.fieldBuffer = std::shared_ptr< float_X >{
-                new float_X[ mThreadParams.window.localDimensions.size
-                                 .productOfComponents() ],
-                []( float_X * ptr ) { delete[] ptr; }
-            };
+            auto size = mThreadParams.window.localDimensions.size
+                                 .productOfComponents();
+            if( size > 0 )
+                mThreadParams.fieldBuffer = std::shared_ptr< float_X >{
+                    new float_X[ size ],
```

Comment:
Forgot to ask during your talk: is there an issue with using `std::vector` instead of C-style array? (and potentially `.data()` to get a raw pointer to its elements).

Reply:
This should be no problem. The openPMD API explicitly allows to use [contiguous containers](https://github.com/openPMD/openPMD-api/blob/dev/include/openPMD/RecordComponent.hpp#L333) and implements this by [mocking a shared pointer](https://github.com/openPMD/openPMD-api/blob/dev/include/openPMD/auxiliary/ShareRaw.hpp) via `.data()`. 
Would using a `std::vector` be preferrable?

Reply:
I like `std::vector` a little bit more. Of course, in this case, it is not such a big deal since even with the raw pointers you take care that it is properly deleted.

Reply:
Yep, always prefer smart-ptrs and std::vector over C-style arrays.
openPMD-api support , such as `std::array` and `std::vector` for that purpose, as Franz already mentioned :)
https://github.com/openPMD/openPMD-api/blob/dev/include/openPMD/auxiliary/ShareRaw.hpp
https://github.com/openPMD/openPMD-api/pull/386

Reply:
As mentioned further down the thread, I've moved it to `std::vector` due to easier resizing.